docling_core/types/doc/base.py

```diff
@@ -176,3 +263,116 @@ def to_top_left_origin(self, page_height: float) -> "BoundingBox":
+        ):
+            return (
+                (self.b <= other.b and other.b < self.t)
+                or (self.b <= other.t and other.t < self.t)
+                or (other.b <= self.b and self.b < other.t)
+                or (other.b <= self.t and self.t < other.t)
+            )
+        else:
+            raise ValueError("BoundingBox is not BOTTOMLEFT")
+
+        return False
+
+    def overlaps_vertically_with_iou(self, other: "BoundingBox", iou: float) -> bool:
+        """overlaps_y_with_iou."""
+        if (
+            self.coord_origin == CoordOrigin.BOTTOMLEFT
+            and other.coord_origin == CoordOrigin.BOTTOMLEFT
+        ):
+
+            if self.overlaps_vertically(other=other):
+
+                u0 = min(self.b, other.b)
+                u1 = max(self.t, other.t)
+
+                i0 = max(self.b, other.b)
+                i1 = min(self.t, other.t)
+
+                iou_ = float(i1 - i0) / float(u1 - u0)
+                return (iou_) > iou
+
+            return False
+        else:
+            raise ValueError("BoundingBox is not BOTTOMLEFT")
+
+        return False
+
+    def is_left_of(self, other: "BoundingBox") -> bool:
+        """is_left_of."""
+        return self.r < other.r
+
+    def is_strictly_left_of(self, other: "BoundingBox", eps: float = 0.001) -> bool:
+        """is_strictly_left_of."""
+        return (self.r + eps) < other.l
+
+    def is_above_of(self, other: "BoundingBox") -> bool:
+        """is_above."""
+        if (
+            self.coord_origin == CoordOrigin.BOTTOMLEFT
+            and other.coord_origin == CoordOrigin.BOTTOMLEFT
+        ):
+            return self.t > other.t
+        else:
+            raise ValueError("BoundingBox is not BOTTOMLEFT")
+
+        return False
+
+    def is_strictly_above_of(self, other: "BoundingBox", eps: float = 0.001) -> bool:
+        """is_strictly_above."""
+        if (
+            self.coord_origin == CoordOrigin.BOTTOMLEFT
+            and other.coord_origin == CoordOrigin.BOTTOMLEFT
+        ):
+            return (self.b + eps) > other.t
+        else:
+            raise ValueError("BoundingBox is not BOTTOMLEFT")
+
+        return False
+
+    def is_horizontally_connected(
+        self, elem_i: "BoundingBox", elem_j: "BoundingBox"
+    ) -> bool:
+        """is_horizontally_connected."""
+        if (
+            self.coord_origin == CoordOrigin.BOTTOMLEFT
+            and elem_i.coord_origin == CoordOrigin.BOTTOMLEFT
+            and elem_j.coord_origin == CoordOrigin.BOTTOMLEFT
+        ):
+
+            min_ij: float = min(elem_i.b, elem_j.b)
+            max_ij: float = max(elem_i.t, elem_j.t)
+
+            if self.b < max_ij and min_ij < self.t:  # overlap_y
+                return False
+
+            if self.l < elem_i.r and elem_j.l < self.r:
+                return True
+
+            return False
+        else:
+            raise ValueError("BoundingBox is not BOTTOMLEFT")
```

Comment:
This error looks like it should not exist. The only error i can legitimately see is "The coordinate origin of this BoundingBox and other BoundingBox is not the same". For all other cases the geometric functions must simply work.

Reply:
done